Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -113,10 +113,10 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
             }
             didReceiveStartCommand = true
             needsViewUpdate = true
-        case let command as RUMStartViewCommand where command.identity !== identity:
+        case let command as RUMStartViewCommand where !identity.equals(command.identity):
             isActiveView = false
             needsViewUpdate = true // sanity update (in case if the user forgets to end this View)
-        case let command as RUMStopViewCommand where command.identity === identity:
+        case let command as RUMStopViewCommand where identity.equals(command.identity):
```

Comment:
we can add a case or if clause to handle when identity view controller becomes nil
as we keep a weak ref to VC, its `ViewScope` will stay forever if its identity `VC` is deallocated

something like that?
```swift
if self.identity.object == nil && self.identity.value == nil { stopView() }
```

Reply:
This is already covered by closing the scope if another scope is started:
```swift
case let command as RUMStartViewCommand where !identity.equals(command.identity):
   isActiveView = false
   needsViewUpdate = true // sanity update (in case if the user forgets to end this View)
```

Also, if the session expires, before we re-create the scope for last `identity` we first check if the identity still exists:
```swift
// RUMSessionScope

guard let expiredViewIdentifiable = expiredView.identity.identifiable else {
    return nil // if the underlying identifiable (`UIVIewController`) no longer exists, skip transferring its scope
}
```
I think we already have enough measures for keeping it consistent. WDYT @buranmert?